### PR TITLE
Add-user-auth

### PR DESCRIPTION
**Pull request by UtsavK55**
Add user authentication

### CODE REVIEW

File: src/constants/index.ts [hunk 1]
```diff
@@ -22,3 +24,9 @@ export const ROUTES = {
 } as const;
 
 export const tokenRegex = /#token=([A-Za-z0-9]+)/;
+
+export const logoImg = require('@assets/logo.png');
```
Comment (adiTrivsme) [inline src/constants/index.ts]: define Image constants separately

Comment (adiTrivsme) [reply src/constants/index.ts]: const IMAGES = {
   "logoImg": require('@assets/logo.png'),
   ...
}

File: src/contexts/Loginprovider.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+export const UserLoginProvider: React.FC<{children: ReactNode}> = ({
+  children,
+}) => {
+  const [token, setToken] = useState<string>('');
+
+  const fetchUsers = async () => {
+    const token = await getData(STORAGE_KEYS.TOKEN);
+    setToken(token || '');
+  };
+  useEffect(() => {
```
Comment (adiTrivsme) [inline src/contexts/Loginprovider.tsx:30]: add blank line

File: src/hooks/useTheme.ts [hunk 3]
```diff
@@ -12,3 +12,12 @@ const useTheme = () => {
 };
 
 export default useTheme;
+
+export const createThemedStyles = (
+  styleFunction: (theme: ThemeContext) => Record<string, any>,
+) => {
+  return () => {
+    const theme = useTheme();
```
Comment (adiTrivsme) [inline src/hooks/useTheme.ts:20]: is this obeying all the Hook rules?

Comment (UtsavK55) [reply src/hooks/useTheme.ts:20]: Yes

File: src/constants/index.ts [hunk 4]
```diff
@@ -22,3 +24,11 @@ export const ROUTES = {
 } as const;
 
 export const tokenRegex = /#token=([A-Za-z0-9]+)/;
+
+export const IMAGES = {
```
Comment (adiTrivsme) [inline src/constants/index.ts:28]: it would be nice if you define it on a dedicated images file

**Review by adiTrivsme — APPROVED** [review]

**Pull request merged**